### ISSUE TITLE
Seg fault in version 2.4.1d

### ISSUE THREAD

**Issue by AlexFinkel** [alexdobin/STAR#68]
I'm getting the following stacktrace after entering the mapping phase with STAR:

No | Address            | Function | Text Location
----|--------------------- |-------------|-------------------------
#0 | 0x00000000004fe8a2 | compareSeqToGenome | SuffixArrayFuns.cpp:27
#1 | 0x0000000003c3b9af | FailureSignalHandler | base/process_state.cc:1433
#2 | 0x00007fb15bf7933f | <no funcname> | .../nptl/sysdeps/pthread/funlockfile.c:30
#3 | 0x00000000004fef64 | maxMappableLength | SuffixArrayFuns.cpp:138
#4 | 0x00000000005171f2 | ReadAlign::maxMappableLength2strands | ReadAlign_maxMappableLength2strands.cpp:82
#5 | 0x000000000050d8a4 | ReadAlign::mapOneRead | ReadAlign_mapOneRead.cpp:60
#6 | 0x00000000005071d6 | ReadAlign::oneRead | ReadAlign_oneRead.cpp:70
#7 | 0x0000000000505b50 | ReadAlignChunk::mapChunk | ReadAlignChunk_mapChunk.cpp:25
#8 | 0x00000000004f77cc | ReadAlignChunk::processChunks | ReadAlignChunk_processChunks.cpp:144
#9 | 0x00000000004e580c | mapThreadsSpawn | mapThreadsSpawn.cpp:19
#10 | 0x00000000004a3625 | main | STAR.cpp:256

```
uint compareSeqToGenome(char** s2, uint S, uint N, uint L, char* g, PackedArray& SA, uint iSA, bool dirR, bool& compRes, Parameters* P) {
    // compare s to g, find the maximum identity length
    // s2[0] read sequence; s2[1] complementary sequence
    // S position to start search from in s2[0],s2[1]
    //dirR forward or reverse direction search on read sequence
    
    register int64 ii; 
    
    uint SAstr=SA[iSA];
    bool dirG = (SAstr>>P->GstrandBit) == 0; //forward or reverse strand of the genome
    SAstr &= P->GstrandMask;
    
    
    if (dirR && dirG) {//forward on read, forward on genome
        char* s  = s2[0] + S + L;
        g += SAstr + L;
        for (ii=0;(uint) ii < N-L; ii++)
        {   
            if (s[ii]!=g[ii])   <-- seg fault occurs here
```

This code is kind of hard to follow - is it possible that ii could be out of bounds for s or g?

The command run was:

    rna-star --genomeDir=Homo_sapiens.GRCh38.dna.primary_assembly.index.2 --outFileNamePrefix=output/ --outTmpDir=./_STARtmp --readFilesIn=fastq_reads/SRR062634.filt.fastq --runThreadN=1

(I was just testing it out on publicly available data).

There's a few local modifications in my source to do with file handling but I don't think any of them are relevant. I may have also deleted some commented-out code here and there, so it's possible that the line numbers don't match perfectly.

**Comment by alexdobin** [comment]
Hi Alex,

could you please post the Log.out file?

Cheers
Alex

**Comment by AlexFinkel** [comment]
Pasted Log.out and Log.progress.out below. I replaced the long absolute file paths with relative paths but otherwise left things untouched.

output/Log.out:

```
STAR version=STAR_2.4.1d
STAR compilation time,server,dir=google3_blaze
##### DEFAULT parameters:
versionSTAR                       20201
versionGenome                     20101   20200   
parametersFiles                   -   
sysShell                          -
runMode                           alignReads
runThreadN                        1
runDirPerm                        User_RWX
genomeDir                         ./GenomeDir/
genomeLoad                        NoSharedMemory
genomeFastaFiles                  -   
genomeSAindexNbases               14
genomeChrBinNbits                 18
genomeSAsparseD                   1
readFilesIn                       Read1   Read2   
readFilesCommand                  -   
readMatesLengthsIn                NotEqual
readMapNumber                     18446744073709551615
inputBAMfile                      -
bamRemoveDuplicatesType           -
bamRemoveDuplicatesMate2basesN    0
limitGenomeGenerateRAM            31000000000
limitIObufferSize                 150000000
limitOutSAMoneReadBytes           100000
limitOutSJcollapsed               1000000
limitOutSJoneRead                 1000
limitBAMsortRAM                   0
limitSjdbInsertNsj                1000000
outStd                            Log
outReadsUnmapped                  None
outQSconversionAdd                0
outSAMtype                        SAM   
outSAMmode                        Full
outSAMstrandField                 None
outSAMattributes                  Standard   
outSAMunmapped                    None
outSAMorder                       Paired
outSAMprimaryFlag                 OneBestScore
outSAMreadID                      Standard
outSAMmapqUnique                  255
outSAMflagOR                      0
outSAMflagAND                     65535
outSAMattrRGline                  -   
outSAMheaderHD                    -   
outSAMheaderPG                    -   
outSAMheaderCommentFile           -
outBAMcompression                 1
outBAMsortingThreadN              0
outSJfilterReads                  All
outSJfilterCountUniqueMin         3   1   1   1   
outSJfilterCountTotalMin          3   1   1   1   
outSJfilterOverhangMin            30   12   12   12   
outSJfilterDistToOtherSJmin       10   0   5   10   
outSJfilterIntronMaxVsReadN       50000   100000   200000   
outWigType                        None   
outWigStrand                      Stranded   
outWigReferencesPrefix            -
outWigNorm                        RPM   
outFilterType                     Normal
outFilterMultimapNmax             10
outFilterMultimapScoreRange       1
outFilterScoreMin                 0
outFilterScoreMinOverLread        0.66
outFilterMatchNmin                0
outFilterMatchNminOverLread       0.66
outFilterMismatchNmax             10
outFilterMismatchNoverLmax        0.3
outFilterMismatchNoverReadLmax    1
outFilterIntronMotifs             None
clip5pNbases                      0   
clip3pNbases                      0   
clip3pAfterAdapterNbases          0   
clip3pAdapterSeq                  -   
clip3pAdapterMMp                  0.1   
winBinNbits                       16
winAnchorDistNbins                9
winFlankNbins                     4
winAnchorMultimapNmax             50
scoreGap                          0
scoreGapNoncan                    -8
scoreGapGCAG                      -4
scoreGapATAC                      -8
scoreStitchSJshift                1
scoreGenomicLengthLog2scale       -0.25
scoreDelBase                      -2
scoreDelOpen                      -2
scoreInsOpen                      -2
scoreInsBase                      -2
seedSearchLmax                    0
seedSearchStartLmax               50
seedSearchStartLmaxOverLread      1
seedPerReadNmax                   1000
seedPerWindowNmax                 50
seedNoneLociPerWindow             10
seedMultimapNmax                  10000
alignIntronMin                    21
alignIntronMax                    0
alignMatesGapMax                  0
alignTranscriptsPerReadNmax       10000
alignSJoverhangMin                5
alignSJDBoverhangMin              3
alignSplicedMateMapLmin           0
alignSplicedMateMapLminOverLmate    0.66
alignWindowsPerReadNmax           10000
alignTranscriptsPerWindowNmax     100
alignEndsType                     Local
alignSoftClipAtReferenceEnds      Yes
chimSegmentMin                    0
chimScoreMin                      0
chimScoreDropMax                  20
chimScoreSeparation               10
chimScoreJunctionNonGTAG          -1
chimJunctionOverhangMin           20
chimOutType                       SeparateSAMold
sjdbFileChrStartEnd               -   
sjdbGTFfile                       -
sjdbGTFchrPrefix                  -
sjdbGTFfeatureExon                exon
sjdbGTFtagExonParentTranscript    transcript_id
sjdbGTFtagExonParentGene          gene_id
sjdbOverhang                      100
sjdbScore                         2
sjdbInsertSave                    Basic
quantMode                         -   
quantTranscriptomeBAMcompression    1
quantTranscriptomeBan             IndelSoftclipSingleend
twopass1readsN                    18446744073709551615
twopassMode                       None
##### Command Line:
./rna-star-dbg --genomeDir=rna/Homo_sapiens.GRCh38.dna.primary_assembly.index.2 --outFileNamePrefix=rna/output/ --outTmpDir=./_STARtmp --readFilesIn=rna/fastq_reads/SRR062634.filt.fastq --runThreadN=1
##### Initial USER parameters from Command Line:
outFileNamePrefix                 rna/output/
outTmpDir                         ./_STARtmp
###### All USER parameters from Command Line:
genomeDir                     rna/Homo_sapiens.GRCh38.dna.primary_assembly.index.2     ~RE-DEFINED
outFileNamePrefix             rna/output/     ~RE-DEFINED
outTmpDir                     ./_STARtmp     ~RE-DEFINED
readFilesIn                   rna/fastq_reads/SRR062634.filt.fastq        ~RE-DEFINED
runThreadN                    1     ~RE-DEFINED
##### Finished reading parameters from all sources

##### Final user re-defined parameters-----------------:
runThreadN                        1
genomeDir                         rna/Homo_sapiens.GRCh38.dna.primary_assembly.index.2
readFilesIn                       rna/fastq_reads/SRR062634.filt.fastq   
outFileNamePrefix                 rna/output/
outTmpDir                         ./_STARtmp

-------------------------------
##### Final effective command line:
./rna-star-dbg   --runThreadN 1   --genomeDir rna/Homo_sapiens.GRCh38.dna.primary_assembly.index.2   --readFilesIn rna/fastq_reads/SRR062634.filt.fastq      --outFileNamePrefix rna/output/   --outTmpDir ./_STARtmp

##### Final parameters after user input--------------------------------:
versionSTAR                       20201
versionGenome                     20101   20200   
parametersFiles                   -   
sysShell                          -
runMode                           alignReads
runThreadN                        1
runDirPerm                        User_RWX
genomeDir                         rna/Homo_sapiens.GRCh38.dna.primary_assembly.index.2
genomeLoad                        NoSharedMemory
genomeFastaFiles                  -   
genomeSAindexNbases               14
genomeChrBinNbits                 18
genomeSAsparseD                   1
readFilesIn                       rna/fastq_reads/SRR062634.filt.fastq   
readFilesCommand                  -   
readMatesLengthsIn                NotEqual
readMapNumber                     18446744073709551615
inputBAMfile                      -
bamRemoveDuplicatesType           -
bamRemoveDuplicatesMate2basesN    0
limitGenomeGenerateRAM            31000000000
limitIObufferSize                 150000000
limitOutSAMoneReadBytes           100000
limitOutSJcollapsed               1000000
limitOutSJoneRead                 1000
limitBAMsortRAM                   0
limitSjdbInsertNsj                1000000
outFileNamePrefix                 rna/output/
outTmpDir                         ./_STARtmp
outStd                            Log
outReadsUnmapped                  None
outQSconversionAdd                0
outSAMtype                        SAM   
outSAMmode                        Full
outSAMstrandField                 None
outSAMattributes                  Standard   
outSAMunmapped                    None
outSAMorder                       Paired
outSAMprimaryFlag                 OneBestScore
outSAMreadID                      Standard
outSAMmapqUnique                  255
outSAMflagOR                      0
outSAMflagAND                     65535
outSAMattrRGline                  -   
outSAMheaderHD                    -   
outSAMheaderPG                    -   
outSAMheaderCommentFile           -
outBAMcompression                 1
outBAMsortingThreadN              0
outSJfilterReads                  All
outSJfilterCountUniqueMin         3   1   1   1   
outSJfilterCountTotalMin          3   1   1   1   
outSJfilterOverhangMin            30   12   12   12   
outSJfilterDistToOtherSJmin       10   0   5   10   
outSJfilterIntronMaxVsReadN       50000   100000   200000   
outWigType                        None   
outWigStrand                      Stranded   
outWigReferencesPrefix            -
outWigNorm                        RPM   
outFilterType                     Normal
outFilterMultimapNmax             10
outFilterMultimapScoreRange       1
outFilterScoreMin                 0
outFilterScoreMinOverLread        0.66
outFilterMatchNmin                0
outFilterMatchNminOverLread       0.66
outFilterMismatchNmax             10
outFilterMismatchNoverLmax        0.3
outFilterMismatchNoverReadLmax    1
outFilterIntronMotifs             None
clip5pNbases                      0   
clip3pNbases                      0   
clip3pAfterAdapterNbases          0   
clip3pAdapterSeq                  -   
clip3pAdapterMMp                  0.1   
winBinNbits                       16
winAnchorDistNbins                9
winFlankNbins                     4
winAnchorMultimapNmax             50
scoreGap                          0
scoreGapNoncan                    -8
scoreGapGCAG                      -4
scoreGapATAC                      -8
scoreStitchSJshift                1
scoreGenomicLengthLog2scale       -0.25
scoreDelBase                      -2
scoreDelOpen                      -2
scoreInsOpen                      -2
scoreInsBase                      -2
seedSearchLmax                    0
seedSearchStartLmax               50
seedSearchStartLmaxOverLread      1
seedPerReadNmax                   1000
seedPerWindowNmax                 50
seedNoneLociPerWindow             10
seedMultimapNmax                  10000
alignIntronMin                    21
alignIntronMax                    0
alignMatesGapMax                  0
alignTranscriptsPerReadNmax       10000
alignSJoverhangMin                5
alignSJDBoverhangMin              3
alignSplicedMateMapLmin           0
alignSplicedMateMapLminOverLmate    0.66
alignWindowsPerReadNmax           10000
alignTranscriptsPerWindowNmax     100
alignEndsType                     Local
alignSoftClipAtReferenceEnds      Yes
chimSegmentMin                    0
chimScoreMin                      0
chimScoreDropMax                  20
chimScoreSeparation               10
chimScoreJunctionNonGTAG          -1
chimJunctionOverhangMin           20
chimOutType                       SeparateSAMold
sjdbFileChrStartEnd               -   
sjdbGTFfile                       -
sjdbGTFchrPrefix                  -
sjdbGTFfeatureExon                exon
sjdbGTFtagExonParentTranscript    transcript_id
sjdbGTFtagExonParentGene          gene_id
sjdbOverhang                      100
sjdbScore                         2
sjdbInsertSave                    Basic
quantMode                         -   
quantTranscriptomeBAMcompression    1
quantTranscriptomeBan             IndelSoftclipSingleend
twopass1readsN                    18446744073709551615
twopassMode                       None
----------------------------------------

Finished loading and checking parameters
Reading genome generation parameters:
versionGenome                 20201        ~RE-DEFINED
genomeFastaFiles              rna/Homo_sapiens.GRCh38.dna.primary_assembly.fa        ~RE-DEFINED
genomeSAindexNbases           14     ~RE-DEFINED
genomeChrBinNbits             18     ~RE-DEFINED
genomeSAsparseD               1     ~RE-DEFINED
sjdbOverhang                  0     ~RE-DEFINED
sjdbFileChrStartEnd           -        ~RE-DEFINED
sjdbGTFfile                   -     ~RE-DEFINED
sjdbGTFchrPrefix              -     ~RE-DEFINED
sjdbGTFfeatureExon            exon     ~RE-DEFINED
sjdbGTFtagExonParentTranscripttranscript_id     ~RE-DEFINED
sjdbGTFtagExonParentGene      gene_id     ~RE-DEFINED
sjdbInsertSave                Basic     ~RE-DEFINED
Genome version is compatible with current STAR version
--sjdbOverhang = 0 taken from the generated genome
Started loading the genome: Tue Aug 25 09:28:16 2015

checking Genome sizefile size: 3138387968 bytes; state: good=1 eof=0 fail=0 bad=0
checking SA sizefile size: 24303254803 bytes; state: good=1 eof=0 fail=0 bad=0
checking SAindex sizefile size: 1565873619 bytes; state: good=1 eof=0 fail=0 bad=0
Read from SAindex: genomeSAindexNbases=14  nSAi=357913940
nGenome=3138387968;  nSAbyte=24303254803
GstrandBit=32   SA number of indices=5891698134
Shared memory is not used for genomes. Allocated a private copy of the genome.
Genome file size: 3138387968 bytes; state: good=1 eof=0 fail=0 bad=0
Loading Genome ... done! state: good=1 eof=0 fail=0 bad=0; loaded 3138387968 bytes
SA file size: 24303254806 bytes; state: good=1 eof=0 fail=0 bad=0
Loading SA ... done! state: good=0 eof=1 fail=1 bad=0; loaded 24303254803 bytes
Loading SAindex ... done: 1565873619 bytes
Finished loading the genome: Tue Aug 25 12:08:39 2015

Number of real (reference) chromosmes= 194
1	1	248956422	0
2	10	133797422	249036800
3	11	135086622	382992384
4	12	133275309	518258688
5	13	114364328	651689984
6	14	107043718	766246912
7	15	101991189	873463808
8	16	90338345	975699968
9	17	83257441	1066139648
10	18	80373285	1149501440
11	19	58617616	1229979648
12	2	242193529	1288699904
13	20	64444167	1530920960
14	21	46709983	1595408384
15	22	50818468	1642332160
16	3	198295559	1693188096
17	4	190214555	1891631104
18	5	181538259	2081947648
19	6	170805979	2263613440
20	7	159345973	2434531328
21	8	145138636	2593914880
22	9	138394717	2739142656
23	MT	16569	2877554688
24	X	156040895	2877816832
25	Y	57227415	3034054656
26	KI270728.1	1872759	3091464192
27	KI270727.1	448248	3093561344
28	KI270442.1	392061	3094085632
29	KI270729.1	280839	3094609920
30	GL000225.1	211173	3095134208
31	KI270743.1	210658	3095396352
32	GL000008.2	209709	3095658496
33	GL000009.2	201709	3095920640
34	KI270747.1	198735	3096182784
35	KI270722.1	194050	3096444928
36	GL000194.1	191469	3096707072
37	KI270742.1	186739	3096969216
38	GL000205.2	185591	3097231360
39	GL000195.1	182896	3097493504
40	KI270736.1	181920	3097755648
41	KI270733.1	179772	3098017792
42	GL000224.1	179693	3098279936
43	GL000219.1	179198	3098542080
44	KI270719.1	176845	3098804224
45	GL000216.2	176608	3099066368
46	KI270712.1	176043	3099328512
47	KI270706.1	175055	3099590656
48	KI270725.1	172810	3099852800
49	KI270744.1	168472	3100114944
50	KI270734.1	165050	3100377088
51	GL000213.1	164239	3100639232
52	GL000220.1	161802	3100901376
53	KI270715.1	161471	3101163520
54	GL000218.1	161147	3101425664
55	KI270749.1	158759	3101687808
56	KI270741.1	157432	3101949952
57	GL000221.1	155397	3102212096
58	KI270716.1	153799	3102474240
59	KI270731.1	150754	3102736384
60	KI270751.1	150742	3102998528
61	KI270750.1	148850	3103260672
62	KI270519.1	138126	3103522816
63	GL000214.1	137718	3103784960
64	KI270708.1	127682	3104047104
65	KI270730.1	112551	3104309248
66	KI270438.1	112505	3104571392
67	KI270737.1	103838	3104833536
68	KI270721.1	100316	3105095680
69	KI270738.1	99375	3105357824
70	KI270748.1	93321	3105619968
71	KI270435.1	92983	3105882112
72	GL000208.1	92689	3106144256
73	KI270538.1	91309	3106406400
74	KI270756.1	79590	3106668544
75	KI270739.1	73985	3106930688
76	KI270757.1	71251	3107192832
77	KI270709.1	66860	3107454976
78	KI270746.1	66486	3107717120
79	KI270753.1	62944	3107979264
80	KI270589.1	44474	3108241408
81	KI270726.1	43739	3108503552
82	KI270735.1	42811	3108765696
83	KI270711.1	42210	3109027840
84	KI270745.1	41891	3109289984
85	KI270714.1	41717	3109552128
86	KI270732.1	41543	3109814272
87	KI270713.1	40745	3110076416
88	KI270754.1	40191	3110338560
89	KI270710.1	40176	3110600704
90	KI270717.1	40062	3110862848
91	KI270724.1	39555	3111124992
92	KI270720.1	39050	3111387136
93	KI270723.1	38115	3111649280
94	KI270718.1	38054	3111911424
95	KI270317.1	37690	3112173568
96	KI270740.1	37240	3112435712
97	KI270755.1	36723	3112697856
98	KI270707.1	32032	3112960000
99	KI270579.1	31033	3113222144
100	KI270752.1	27745	3113484288
101	KI270512.1	22689	3113746432
102	KI270322.1	21476	3114008576
103	GL000226.1	15008	3114270720
104	KI270311.1	12399	3114532864
105	KI270366.1	8320	3114795008
106	KI270511.1	8127	3115057152
107	KI270448.1	7992	3115319296
108	KI270521.1	7642	3115581440
109	KI270581.1	7046	3115843584
110	KI270582.1	6504	3116105728
111	KI270515.1	6361	3116367872
112	KI270588.1	6158	3116630016
113	KI270591.1	5796	3116892160
114	KI270522.1	5674	3117154304
115	KI270507.1	5353	3117416448
116	KI270590.1	4685	3117678592
117	KI270584.1	4513	3117940736
118	KI270320.1	4416	3118202880
119	KI270382.1	4215	3118465024
120	KI270468.1	4055	3118727168
121	KI270467.1	3920	3118989312
122	KI270362.1	3530	3119251456
123	KI270517.1	3253	3119513600
124	KI270593.1	3041	3119775744
125	KI270528.1	2983	3120037888
126	KI270587.1	2969	3120300032
127	KI270364.1	2855	3120562176
128	KI270371.1	2805	3120824320
129	KI270333.1	2699	3121086464
130	KI270374.1	2656	3121348608
131	KI270411.1	2646	3121610752
132	KI270414.1	2489	3121872896
133	KI270510.1	2415	3122135040
134	KI270390.1	2387	3122397184
135	KI270375.1	2378	3122659328
136	KI270420.1	2321	3122921472
137	KI270509.1	2318	3123183616
138	KI270315.1	2276	3123445760
139	KI270302.1	2274	3123707904
140	KI270518.1	2186	3123970048
141	KI270530.1	2168	3124232192
142	KI270304.1	2165	3124494336
143	KI270418.1	2145	3124756480
144	KI270424.1	2140	3125018624
145	KI270417.1	2043	3125280768
146	KI270508.1	1951	3125542912
147	KI270303.1	1942	3125805056
148	KI270381.1	1930	3126067200
149	KI270529.1	1899	3126329344
150	KI270425.1	1884	3126591488
151	KI270396.1	1880	3126853632
152	KI270363.1	1803	3127115776
153	KI270386.1	1788	3127377920
154	KI270465.1	1774	3127640064
155	KI270383.1	1750	3127902208
156	KI270384.1	1658	3128164352
157	KI270330.1	1652	3128426496
158	KI270372.1	1650	3128688640
159	KI270548.1	1599	3128950784
160	KI270580.1	1553	3129212928
161	KI270387.1	1537	3129475072
162	KI270391.1	1484	3129737216
163	KI270305.1	1472	3129999360
164	KI270373.1	1451	3130261504
165	KI270422.1	1445	3130523648
166	KI270316.1	1444	3130785792
167	KI270340.1	1428	3131047936
168	KI270338.1	1428	3131310080
169	KI270583.1	1400	3131572224
170	KI270334.1	1368	3131834368
171	KI270429.1	1361	3132096512
172	KI270393.1	1308	3132358656
173	KI270516.1	1300	3132620800
174	KI270389.1	1298	3132882944
175	KI270466.1	1233	3133145088
176	KI270388.1	1216	3133407232
177	KI270544.1	1202	3133669376
178	KI270310.1	1201	3133931520
179	KI270412.1	1179	3134193664
180	KI270395.1	1143	3134455808
181	KI270376.1	1136	3134717952
182	KI270337.1	1121	3134980096
183	KI270335.1	1048	3135242240
184	KI270378.1	1048	3135504384
185	KI270379.1	1045	3135766528
186	KI270329.1	1040	3136028672
187	KI270419.1	1029	3136290816
188	KI270336.1	1026	3136552960
189	KI270312.1	998	3136815104
190	KI270539.1	993	3137077248
191	KI270385.1	990	3137339392
192	KI270423.1	981	3137601536
193	KI270392.1	971	3137863680
194	KI270394.1	970	3138125824
alignIntronMax=alignMatesGapMax=0, the max intron size will be approximately determined by (2^winBinNbits)*winAnchorDistNbins=589824
```

output/Log.progress.out:
```
           Time    Speed        Read     Read   Mapped   Mapped   Mapped   Mapped Unmapped Unmapped Unmapped Unmapped
                    M/hr      number   length   unique   length   MMrate    multi   multi+       MM    short    other
```

**Comment by alexdobin** [comment]
Hi Alex,

I believe this problem was caused a bug in the genome generation without annotation. I have fixed it in the patch on github-master: https://github.com/alexdobin/STAR/archive/master.zip
Please try it out and let me know if it solved the problem.

Cheers
Alex

**Comment by AlexFinkel** [comment]
I finally got around to loading the latest code and re-running star - it works! No seg faults in sight :) Thank you for the patch.
